File: src/class_pubsub.cpp
#include <ros/ros.h>
#include <std_msgs/Int64.h>
#include <std_srvs/SetBool.h>

class NumberCounter {

  private:
    int counter;
    ros::Publisher pub;
    ros::Subscriber number_subscriber;
  
  public:
    NumberCounter(ros::NodeHandle *nh) {
      counter = 0;
      pub = nh->advertise<std_msgs::Int64>("/number_count", 10);    
      number_subscriber = nh->subscribe("/number", 1000, &NumberCounter::callback_number, this);
    }
  
    void callback_number(const std_msgs::Int64& msg) {
      counter += msg.data;
      std_msgs::Int64 new_msg;
      new_msg.data = counter;
      pub.publish(new_msg);
    }
};

int main (int argc, char **argv)
{
  ros::init(argc, argv, "number_counter");
  ros::NodeHandle nh;
  NumberCounter nc = NumberCounter(&nh);
  ros::spin();
}

File: src/class_pubsub.py
#!/usr/bin/env python3
import rospy
from std_msgs.msg import String
from std_msgs.msg import Float64

class NumberCounter:
  
  def __init__(self):
    self.count = 0
    self.resultado = 0
    self.sub = rospy.Subscriber("counter", Float64, self.funcion)
    self.pub = rospy.Publisher("counter_pair", String, queue_size=1)

  def funcion(self, msg): # Tipo callback
    self.resultado = msg.data

if __name__ == "__main__":

  rospy.init_node('publishersubscriber')
  class_pub = NumberCounter()
  rate = rospy.Rate(2) # 2 Hz
  
  while not rospy.is_shutdown():
    
    message = String()
    
    if (class_pub.resultado % 2 == 0):
      message.data = "Valor par"
    else:
      message.data = "Valor impar"
    
    class_pub.pub.publish(message)
    rate.sleep()
